Makefile

```diff
@@ -30,8 +30,16 @@ submodule submodules: $(submodule_paths)
 ## End Git Submodules
 
 ## Begin Web
-dist: package.json pnpm-lock.yaml
-	pnpm i
+prepare-pnpm:
+	@if which corepack > /dev/null; then \
+		corepack enable; \
+		corepack prepare pnpm@latest --activate; \
```

Comment:
I'm not sure it's a good idea to always use the latest version of pnpm, maybe we choose to add the `packageManager` field to `package.json` to use a fixed version of pnpm?

Reply:
Nerver heard of this before. Thank you for your sharing. This totally makes sense.

After some reading, I realize it's a currently in an experimental state, but it wont be stopping us to use this feature.

See https://github.com/nodejs/node/blob/main/doc/api/packages.md?plain=1#L1181

If you want to collaborate on this, feel free to push changes to this branch

Reply:
I do not think we should introduce anything `permanent` package in Makefile, especially to the systematic path directory.